File: Insertion Sort for Singly Linked List/Insertion Sort for Singly Linked List.py
class Solution:
    def insertionSort(self, head):
        #code here
        
        if not head or not head.next:
            return head
            
        temp = Node(-1)
        temp.next = head
            
        current = head
        
        while(current.next):
            if current.next.data < current.data:
                insert = current.next
                current.next = current.next.next
                
                pt = temp
                while(pt and pt.next.data) <(insert.data):
                    pt = pt.next
                
                insert.next = pt.next
                pt.next = insert
            
            else:
                current = current.next
                
        return temp.next        
        
        

File: Insertion Sort for Singly Linked List/Insertion Sort for Singly Linked List.cpp
  if(head_ref==NULL ||head_ref->next==NULL )
        return head_ref;
        Node *head=head_ref;
        Node *cur=head_ref;
        while(cur!=NULL)
        {
            while(head!=cur)
            {
                if(head->data>cur->data)
                {
                    int d=head->data;
                    head->data=cur->data;
                    cur->data=d;
                }
                head=head->next;
            }
            head=head_ref;
            cur=cur->next;
        }
        return head_ref;
